testutil/ciao-down/README.md

```diff
@@ -10,39 +12,336 @@ Then simply type
 
 ```
 go get github.com/01org/ciao/testutil/ciao-down
-$GOPATH/bin/ciao-down prepare
+$GOPATH/bin/ciao-down prepare --vmtype=xenial
+```
+
+to create a new Ubuntu 16.04 VM.  ciao-down will install some needed
+dependencies on your local PC such as qemu and xorriso. It will then
+download an Ubuntu Cloud Image and create a VM based on this image.
+It will boot the VM, create an account for you, with the same
+name as your account on your host, and optionally update the default
+packages.
+
+Once it's finished you'll be able to connect to the the VM via SSH
+using the following command.
+
+```
+ciao-down connect
+```
+
+The command above assumes that either $GOPATH/bin or ~/go/bin is on
+your PATH.
+
+ciao-down will cache the Ubuntu image locally so the next time you
+create another xenial based you won't have to wait very long. It also
+knows about HTTP proxies and will mirror your host computer's proxy
+settings inside the VMs it creates.
+
+You can delete the VM you've just created by calling,
+
+```
+ciao-down delete
+```
+
+## Workloads
+
+Each VM is created from a workload. A workload is a text file which contains
+a set of instructions for initialising the VM.  A workload, among other things,
+can specify:
+
+- The hostname of the VM.
+- The resources to be consumed by the VM
+- The base image from which the VM is to be created
+- The folders that should be shared between the host and the VM
+- An annotated cloud-init file that contains the set of instructions
+  to run on the first boot of the VM.  This file is used to create
+  user accounts, install packages and configure the VM.
+
+ciao-down ships with a number of workloads for creating VMs based on standard images,
+such as Ubuntu 16.04 and Fedora 25.  Users are also free to create their own workloads.
+Standard workloads are stored in $GOPATH/src/github.com/01org/ciao/testutil/ciao-down/workloads.
+User created workloads are stored in ~/.ciao-down/workloads.  ciao-down always checks the
+~/.ciao-down/workloads directory first so if a workload exists in both directories
+with the same name, ciao-down will use the workload in ~/.ciao-down/workloads.
+
+You can tell ciao-down which workload to use when creating the VM with the
+prepare command.  This is done via the --vmtype option.  You specify the file name
```

Comment:
Yes, I agree.  The reason it's still called vmtype is that I didn't want to break ciao-down for existing users and scripts.  But I could add a workload option as well which would do the same thing as vmtype.